core/templates/dev/head/services/playthrough-issues-backend-api.service.ts

```diff
@@ -55,7 +55,7 @@ angular.module('oppia').factory('PlaythroughIssuesBackendApiService', [
     };
     return {
       fetchIssues: function(explorationId, explorationVersion) {
-        if (cachedIssues.length !== 0) {
+        if (cachedIssues !== null && cachedIssues.length !== 0) {
```

Comment:
Actually I think this could just be "if (cachedIssues !== null)"?

Reply:
(If the length is 0 then it's still going to be 0 on a refetch (presumably), so why do we need to fetch again?)

Reply:
If the length is 0, is it always be 0 on a refetch? Following this logic, the length would always be same, regardless of the length value, so the fetch would be necessary.

Reply:
Probably not in the same user session. The point is that we want to fetch the data once, and then, after that, just refer to the cached version in the frontend.

So, the logic here should not depend on the length of the array. Either we fetch every single time (because the array may have updated), or we fetch once at the outset and use our cached copy for all subsequent operations. Does that make sense? /cc @brianrodri

Reply:
Yeah, it does make sense. Just to be sure that I get it: fetchIssues should call http.get only once in the same user session?

Reply:
Yes, that is correct.

Reply:
Fetching exactly once sounds perfect to me, so checking only against null makes the most sense.